src/html-splitter.ts

```diff
@@ -148,32 +155,50 @@ class HtmlSplitTransform extends AsyncTransformStream<File, File> {
       const contents = await getFileContents(file);
       const doc = parse5.parse(contents, {locationInfo: true});
       dom5.removeFakeRootElements(doc);
-      const scriptTags = dom5.queryAll(doc, HtmlSplitTransform.isInlineScript);
+      const scriptTags = dom5.queryAll(doc, pred.hasTagName('script'));
+      let moduleScriptIdx = 0;
       for (let i = 0; i < scriptTags.length; i++) {
         const scriptTag = scriptTags[i];
         const source = dom5.getTextContent(scriptTag);
-        const typeAtribute =
+        const typeAttribute =
             dom5.getAttribute(scriptTag, 'type') || 'application/javascript';
-        const extension = extensionsForType[typeAtribute];
+        const extension = extensionsForType[typeAttribute];
         // If we don't recognize the script type attribute, don't split
         // out.
         if (!extension) {
           continue;
         }
 
-        const childFilename =
-            `${osPath.basename(filePath)}_script_${i}.${extension}`;
-        const childPath = osPath.join(osPath.dirname(filePath), childFilename);
-        scriptTag.childNodes = [];
-        dom5.setAttribute(scriptTag, 'src', childFilename);
-        const scriptFile = new File({
-          cwd: file.cwd,
-          base: file.base,
-          path: childPath,
-          contents: new Buffer(source),
-        });
-        this._state.addSplitPath(filePath, childPath);
-        this.push(scriptFile);
+        const isModule = typeAttribute === 'module';
+        const isInline = !dom5.hasAttribute(scriptTag, 'src');
+
+        if (isInline) {
+          const childFilename =
+              `${osPath.basename(filePath)}_script_${i}.${extension}`;
+          const childPath =
+              osPath.join(osPath.dirname(filePath), childFilename);
+          scriptTag.childNodes = [];
+          dom5.setAttribute(scriptTag, 'src', childFilename);
+          dom5.setAttribute(scriptTag, htmlSplitterAttribute, '');
+          const scriptFile = new File({
+            cwd: file.cwd,
+            base: file.base,
+            path: childPath,
+            contents: new Buffer(source),
```

Comment:
FYI, `new Buffer()` is deprecated: https://nodejs.org/api/buffer.html#buffer_new_buffer_buffer
but Buffer.from isn't in Node 4

Reply:
Ok, I guess I don't want to change that in this PR though. These lines changed only because I intended them.